cirq/sim/mux_test.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2019 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Sampling/simulation methods that delegate to appropriate simulators."""
+import collections
+
+import sympy
+
+import cirq
+
+
+def test_run():
+    q = cirq.NamedQubit('q')
+
+    # Unitary.
+    results = cirq.sample(cirq.Circuit.from_ops(cirq.X(q), cirq.measure(q)))
+    assert results.histogram(key=q) == collections.Counter({1: 1})
+
+    # Intermediate measurements.
+    results = cirq.sample(cirq.Circuit.from_ops(
+        cirq.measure(q, key='drop'),
+        cirq.X(q),
+        cirq.measure(q),
+    ))
+    assert results.histogram(key='drop') == collections.Counter({0: 1})
+    assert results.histogram(key=q) == collections.Counter({1: 1})
+
+
+def test_run_sweep():
+    q = cirq.NamedQubit('q')
+    c = cirq.Circuit.from_ops(
+        cirq.X(q),
+        cirq.Z(q)**sympy.Symbol('t'),
```

Comment:
Regardless of the value of t, the Z gate doesn't affect the measurement outcome. Consequently, the values we assert on below are independent of whether or not the sweep is working!

Suggestion: Move symbolic exponent from the Z gate to the X gate, remove Z gate, change sweep to only exercise values 0 and 1 and verify that the measurement returns 0 in the t=0 case and 1 in the t=1 case.

Reply:
Done.